bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/about/AboutFeaturesPage.java

```diff
@@ -61,6 +66,33 @@
  */
 public class AboutFeaturesPage extends ProductInfoPage {
 
+	public class BundleGroupTableLabelProvider extends LabelProvider implements ITableLabelProvider {
+
+		@Override
+		public String getColumnText(Object element, int columnIndex) {
+			if (element instanceof AboutBundleGroupData) {
+				AboutBundleGroupData data = (AboutBundleGroupData) element;
+				switch (columnIndex) {
+				case 0:
+					return data.getProviderName();
+				case 1:
+					return data.getName();
+				case 2:
+					return data.getVersion();
+				case 3:
+					return data.getId();
+				}
+			}
+			return ""; //$NON-NLS-1$
```

Comment:
Just to promote newer Java
```suggestion
			return element instanceof AboutBundleGroupData data ?
				switch (columnIndex) {
					case 0 -> data.getProviderName();
					case 1 -> data.getName();
					case 2 -> data.getVersion();
					case 3 -> data.getId();
					default -> "";  //$NON-NLS-1$
				} : "";  //$NON-NLS-1$
```

Reply:
This syntax is giving me error "data cannot be resolved to a type". so I used the long notation, still making use of the modern switch case semantics. let me know if this is fine with you.

Reply:
OK thanks.
I'm wondering whether the error is normal. Ie does `javac` manage to process it?

Reply:
This is what I get, looks like Javac is unable to recognise it.
<img width="1130" alt="image" src="https://github.com/eclipse-platform/eclipse.platform.ui/assets/122080218/624704ae-7eae-4f35-b285-85d82e9e04ed">

Reply:
Which Eclipse version do you have? Please try to use latest SDK build, there were few fixes in switch area.

Reply:
I am using this version of Eclipse SDK
```
Version: 2024-03 (4.31)
Build id: I20240101-1800
```

did the fix come in the last few days?

Reply:
I'm seeing the same error with Version: 2024-03 (4.31) / Build id: I20240105-1800 , updated about 2 hours ago.
```java
class A {
	String method(Object element, int columnIndex) {
		return element instanceof String data ?
			switch (columnIndex) {
				case 0 -> data;
				case 1 -> data.toUpperCase();
				default -> "";
			} : "";
	}
}
```
makes JDT complain with

> Description	Resource	Path	Location	Type
data cannot be resolved	A.java	/sandboxJava/src/sandboxJava	line 7	Java Problem
data cannot be resolved to a variable	A.java	/sandboxJava/src/sandboxJava	line 6	Java Problem

while `javac A.java` generates a fine `A.class` file.

@elsazac as you were the lucky one spotting the issue, would you mind reporting it to JDT Core?